sdk/cognitivelanguage/azure-ai-language-conversations/azure/ai/language/conversations/aio/_patch.py

```diff
@@ -1,92 +1,324 @@
+
+    :param op_loc: The ``Operation-Location`` header value or URL to parse.
+        If ``None`` or malformed, no ID can be extracted.
+    :type op_loc: Optional[str]
+    :return: The trailing path segment as the operation ID, or ``None`` if not found.
+    :rtype: Optional[str]
     """
+    if not op_loc:
+        return None
+    path = urlparse(op_loc).path.rstrip("/")
+    if "/" not in path:
+        return None
+    return path.rsplit("/", 1)[-1]
+
+
+PollingReturnType_co = TypeVar("PollingReturnType_co", covariant=True)
+
+
+class AnalyzeConversationAsyncLROPoller(AsyncLROPoller[PollingReturnType_co], Generic[PollingReturnType_co]):
+    """Async poller that returns PollingReturnType_co and exposes operation metadata."""
+
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        super().__init__(*args, **kwargs)
+        # populated by your deserialization callback in begin_*_async
+        self._last_state: Optional["AnalyzeConversationOperationState"] = None
+
+    def _record_state_for_details(self, state: "AnalyzeConversationOperationState") -> None:
+        """Internal: update the state used by ``.details``.
+
+        :param state: The latest operation state to expose via ``details``.
+        :type state: AnalyzeConversationOperationState
+        :return: None
+        :rtype: None
+        """
+        self._last_state = state
+
+    @property
+    def details(self) -> Mapping[str, Any]:
+        """Metadata associated with the long-running operation.
+
+        :return: A mapping with keys like ``operation_id`` and, when available,
+            ``status``, ``job_id``, ``display_name``, ``created_date_time``,
+            ``last_updated_date_time``, ``expiration_date_time``, ``statistics``,
+            ``errors``, and ``next_link``.
+        :rtype: Mapping[str, Any]
+        """
+        try:
+            headers = getattr(self.polling_method(), "_initial_response").http_response.headers  # type: ignore[attr-defined]
+            op_loc = headers.get("Operation-Location") or headers.get("operation-location")
+        except (AttributeError, TypeError):
+            # missing attributes in the chain, or headers is not a mapping
+            op_loc = None
+
+        info: Dict[str, Any] = {"operation_id": _parse_operation_id(op_loc)}
+
+        # Enrich from final state if available
+        if self._last_state is not None:
+            s = self._last_state
+            info.update(
+                {
+                    "status": s.status,
+                    "job_id": s.job_id,
+                    "display_name": s.display_name,
+                    "created_date_time": s.created_date_time,
+                    "last_updated_date_time": s.last_updated_date_time,
+                    "expiration_date_time": s.expiration_date_time,
+                    "statistics": s.statistics,
+                    "errors": s.errors,
+                    "next_link": s.next_link,
```

Comment:
We shouldn't need `errors` or `next_link` here - if there's an error it should already get raised. And `next_link` is already handled by the `ItemPaged` that is returned in result.

Reply:
Planning to address in a subsequent beta:
https://github.com/Azure/azure-sdk-for-python/issues/42725